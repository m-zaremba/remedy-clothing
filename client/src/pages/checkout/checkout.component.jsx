import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors';

import {
  CheckoutPageContainer,
  CheckoutHeader,
  HeaderBlock,
  TotalPrice,
  WarningInfo,
} from './checkout.styles';

import CheckoutItem from '../../components/checkout-item/checkout-item.component';
import StripeCheckoutButton from '../../components/stripe-button/stripe-button.component';

const Checkout = ({ cartItems, total }) => {
  return (
    <CheckoutPageContainer>
      <CheckoutHeader>
        <HeaderBlock>
          <span>Product</span>
        </HeaderBlock>
        <HeaderBlock>
          <span>Description</span>
        </HeaderBlock>
        <HeaderBlock>
          <span>Quantity</span>
        </HeaderBlock>
        <HeaderBlock>
          <span>Price</span>
        </HeaderBlock>
        <HeaderBlock>
          <span>Remove</span>
        </HeaderBlock>
      </CheckoutHeader>
      {cartItems.map((cartItem) => (
        <CheckoutItem key={cartItem.id} cartItem={cartItem} />
      ))}
      <TotalPrice>
        <span>TOTAL: ${total}</span>
      </TotalPrice>
      <StripeCheckoutButton price={total} />
      <WarningInfo>
        PLEASE USE THE FOLLOWING CREDENTIALS FOR TEST PAYMENTS
        <br />
        Card number: 4242 4242 4242 4242
        <br />
        EXP: 01/21
        <br />
        CVV: 123
      </WarningInfo>
    </CheckoutPageContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal,
});

Checkout.defaultProps = {
  cartItems: PropTypes.array,
};

Checkout.propTypes = {
  total: PropTypes.number.isRequired,
  cartItems: PropTypes.arrayOf(PropTypes.object),
};

export default connect(mapStateToProps)(Checkout);
