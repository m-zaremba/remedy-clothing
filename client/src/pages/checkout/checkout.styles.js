import styled from 'styled-components';

export const CheckoutPageContainer = styled.div`
  width: 55%;
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 50px auto 0;

  @media screen and (max-width: 800px) {
    width: 96%;
  }
`;

export const CheckoutHeader = styled.div`
  width: 100%;
  padding: 10px 0;
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid darkgrey;
`;

export const HeaderBlock = styled.div`
  text-transform: capitalize;
  width: 23%;

  &:last-child {
    width: 8%;

    /* @media screen and (max-width: 800px) {
      width: unset;
      flex-grow: 1;
    } */
  }

  @media screen and (max-width: 800px) {
    width: unset;
    flex-grow: 1;
  }
`;

export const TotalPrice = styled.div`
  margin-top: 30px;
  margin-left: auto;
  font-size: 36px;

  @media screen and (max-width: 800px) {
    margin-right: auto;
    margin-bottom: 30px;
  }
`;

export const WarningInfo = styled.div`
  text-align: center;
  margin-top: 40px;
  font-size: 24px;
  font-weight: bold;
  color: red;
`;
