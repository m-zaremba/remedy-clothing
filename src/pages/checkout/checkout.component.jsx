import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors';

import './checkout.styles.scss';
import CheckoutItem from '../../components/checkout-item/checkout-item.component';
import StripeCheckoutButton from '../../components/stripe-button/stripe-button.component';

const Checkout = ({ cartItems, total }) => {
  return (
    <div className="checkout-page">
      <div className="checkout-header">
        <div className="header-block">
          <span>Product</span>
        </div>
        <div className="header-block">
          <span>Description</span>
        </div>
        <div className="header-block">
          <span>Quantity</span>
        </div>
        <div className="header-block">
          <span>Price</span>
        </div>
        <div className="header-block">
          <span>Remove</span>
        </div>
      </div>
      {cartItems.map((cartItem) => (
        <CheckoutItem key={cartItem.id} cartItem={cartItem} />
      ))}
      <div className="total">
        <span>TOTAL: ${total}</span>
      </div>
      <StripeCheckoutButton price={total} />
      <div className="test-warning">
        PLEASE USE THE FOLLOWING CREDENTIALS FOR TEST PAYMENTS
        <br />
        Card number: 4242 4242 4242 4242
        <br />
        EXP: 01/21
        <br />
        CVV: 123
      </div>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal,
});

Checkout.defaultProps = {
  cartItems: PropTypes.array,
};

Checkout.propTypes = {
  total: PropTypes.number.isRequired,
  cartItems: PropTypes.arrayOf(PropTypes.object),
};

export default connect(mapStateToProps)(Checkout);
