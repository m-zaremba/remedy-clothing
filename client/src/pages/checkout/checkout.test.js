import React from 'react';
import { shallow } from 'enzyme';
import { Checkout } from './checkout.component';

describe('Checkout', () => {
  let wrapper;
  beforeEach(() => {
    const mockProps = {
      cartItems: [],
      total: 77,
    };

    wrapper = shallow(<Checkout {...mockProps} />);
  });

  it('should render Checkout component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
