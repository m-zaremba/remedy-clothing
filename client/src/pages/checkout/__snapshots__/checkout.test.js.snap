// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout should render Checkout component 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <span>
        Product
      </span>
    </styled.div>
    <styled.div>
      <span>
        Description
      </span>
    </styled.div>
    <styled.div>
      <span>
        Quantity
      </span>
    </styled.div>
    <styled.div>
      <span>
        Price
      </span>
    </styled.div>
    <styled.div>
      <span>
        Remove
      </span>
    </styled.div>
  </styled.div>
  <styled.div>
    <span>
      TOTAL: $
      77
    </span>
  </styled.div>
  <Connect(StripeCheckoutButton)
    price={77}
  />
  <styled.div>
    PLEASE USE THE FOLLOWING CREDENTIALS FOR TEST PAYMENTS
    <br />
    Card number: 4242 4242 4242 4242
    <br />
    EXP: 01/21
    <br />
    CVV: 123
  </styled.div>
</styled.div>
`;
